torchvision/transforms/functional_tensor.py

```diff
@@ -636,3 +620,110 @@ def affine(
         img = torch.round(img).to(out_dtype)
 
     return img
+
+
+def affine(
+        img: Tensor, matrix: List[float], resample: int = 0, fillcolor: Optional[int] = None
+) -> Tensor:
+    """Apply affine transformation on the Tensor image keeping image center invariant.
+
+    Args:
+        img (Tensor): image to be rotated.
+        matrix (list of floats): list of 6 float values representing inverse matrix for affine transformation.
+        resample (int, optional): An optional resampling filter. Default is nearest (=0). Other supported values:
+            bilinear(=2).
+        fillcolor (int, optional): this option is not supported for Tensor input. Fill value for the area outside the
+            transform in the output image is always 0.
+
+    Returns:
+        Tensor: Transformed image.
+    """
+    _interpolation_modes = {
+        0: "nearest",
+        2: "bilinear",
+    }
+
+    _assert_grid_transform_inputs(img, matrix, resample, fillcolor, _interpolation_modes)
+
+    theta = torch.tensor(matrix, dtype=torch.float).reshape(1, 2, 3)
+    shape = img.shape
+    grid = affine_grid(theta, size=(1, shape[-3], shape[-2], shape[-1]), align_corners=False)
+    mode = _interpolation_modes[resample]
+
+    return _apply_grid_transform(img, grid, mode)
+
+
+def _compute_output_size(theta: Tensor, w: int, h: int) -> Tuple[int, int]:
+    point = torch.tensor([0.0, 0.0, 1.0])
+    pts = []
+    for i in [0.0, float(h)]:
+        for j in [0.0, float(w)]:
+            # we need to normalize coordinates according to
+            # [0, s] is mapped [-1, +1] as theta translation parameters are
+            # normalized like that
+            point[1], point[0] = 2.0 * i / w - 1.0, 2.0 * j / h - 1.0
+            new_point = torch.matmul(theta, point)
+            # denormalize back to w, h:
+            new_point = (new_point + 1.0) * torch.tensor([w, h]) / 2.0
+            pts.append(new_point)
+    pts = torch.stack(pts)
+    min_vals, _ = pts.min(dim=0)
+    max_vals, _ = pts.max(dim=0)
+    size = torch.ceil(max_vals) - torch.floor(min_vals)
+    return int(size[0]), int(size[1])
+
+
+def _expanded_affine_grid(theta: Tensor, w: int, h: int, expand: bool = False) -> Tensor:
+    if expand:
+        ow, oh = _compute_output_size(theta, w, h)
+    else:
+        ow, oh = w, h
+    output_grid = torch.zeros(1, oh, ow, 2)
+
+    d = 0.5  # if not align_corners
+
+    point = torch.tensor([0.0, 0.0, 1.0])
+    for i in range(oh):
+        for j in range(ow):
```

Comment:
Maybe, we should optimize grid computation...

Reply:
Yes, we should optimize this otherwise the performance will be very bad.

Can't we do something like (untested and probably has bugs)
```python
i = (torch.arange(oh) + d - oh * 0.5) / (0.5 * h)
j = (torch.arange(ow) + d - ow * 0.5) / (0.5 * w)
ii, jj = torch.meshgrid(i, j)
coords = torch.stack([j, i], dim=-1)
output_grid = torch.matmul(coords, theta.T)
```

Note that I removed the 3rd coordinate of `point`, it might be that it's necessary and we should put it back but this gives the gist of it

Reply:
Thanks for the code snippet. Yes, it is something like that, I was trying to implement it from my side too :)